app/src/main/java/org/mozilla/fenix/home/collections/CollectionMenu.kt

```diff
@@ -0,0 +1,198 @@
+ * Menu shown for a [org.mozilla.fenix.home.collections.Collection].
+ *
+ * @see [DropdownMenu]
+ *
+ * @param showMenu Whether this is currently open and visible to the user.
+ * @param menuItems List of options shown.
+ * @param onDismissRequest Called when the user chooses a menu option or requests to dismiss the menu.
+ */
+@Composable
+fun CollectionMenu(
+    showMenu: Boolean,
+    menuItems: List<CollectionMenuItem>,
+    onDismissRequest: () -> Unit,
+) {
+    DisposableEffect(LocalConfiguration.current.orientation) {
+        onDispose { onDismissRequest() }
+    }
+
+    // DropdownMenu uses the medium shape from MaterialTheme.
+    // Override it's corner radius to be the same 8.dp as in mozac_browser_menu_corner_radius
+    MaterialTheme(shapes = MaterialTheme.shapes.copy(medium = RoundedCornerShape(8.dp))) {
+        DropdownMenu(
+            expanded = showMenu,
+            onDismissRequest = { onDismissRequest() },
+            modifier = Modifier
+                .background(color = FirefoxTheme.colors.layer2),
+        ) {
+            for (item in menuItems) {
+                DropdownMenuItem(
+                    onClick = {
+                        onDismissRequest()
+                        item.onClick()
+                    },
+                ) {
+                    Text(
+                        text = item.title,
+                        color = item.color,
+                        maxLines = 1,
+                        modifier = Modifier
+                            .fillMaxHeight()
+                            .align(Alignment.CenterVertically)
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * A menu item for collections.
+ *
+ * @property title The menu item title.
+ * @property color The color that should be set for the title.
+ * @property onClick Invoked when the user clicks on the menu item.
+ */
+@Immutable
+data class CollectionMenuItem(
+    val title: String,
+    val color: Color,
+    val onClick: () -> Unit,
+)
+
+/**
+ * Constructs and returns the default list of menu options for a [TabCollection].
+ *
+ * @param collection [TabCollection] for which the menu will be shown.
+ * Might serve as an argument for the callbacks for when the user interacts with certain menu options.
+ * @param onCollectionAddTabTapped Invoked when the user chooses to open the tabs from [collection].
+ * @param onRenameCollectionTapped Invoked when the user chooses to rename the [collection].
+ * @param onDeleteCollectionTapped Invoked when the user chooses to add tabs to [collection].
+ * @param onDeleteCollectionTapped Invoked when the user chooses to delete [collection].
+ */
+@Composable
+fun getMenuItems(
+    collection: TabCollection,
+    onCollectionOpenTabsTapped: (TabCollection) -> Unit = {},
+    onRenameCollectionTapped: (TabCollection) -> Unit = {},
+    onCollectionAddTabTapped: (TabCollection) -> Unit = {},
+    onDeleteCollectionTapped: (TabCollection) -> Unit = {},
```

Comment:
Since we always expect the consumer to provide a callback for these, I would remove the default parameter to enforce this.